web/web/api/v1/login.py

```diff
@@ -0,0 +1,164 @@
+from flask import request, jsonify, Blueprint
+from datetime import timedelta
+from werkzeug.security import generate_password_hash
+from python_http_client.exceptions import BadRequestsError
+from marshmallow import ValidationError
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound, MultipleResultsFound
+from flask_jwt_extended import (create_access_token, get_jwt_identity,
+                                create_refresh_token, get_jti)
+from .response_wrapper import ApiResponseWrapper
+from web.database import db
+from web.models.login import Login, LoginSchema
+from web.models.user import User, UserSchema
+from web.redis import revoked_store
+
+
+login_api_bp = Blueprint('login_api_bp', __name__)
+
+
+@login_api_bp.route('/login/<int:login_id>', methods=['GET'])
+def show_login_info(login_id):
+    '''
+    Retrieves one login object
+    '''
+    arw = ApiResponseWrapper()
+    login_schema = LoginSchema(exclude=['password_hash'])
+
+    try:
+        login = Login.query.filter_by(login_id=login_id).one()
+
+    except (MultipleResultsFound, NoResultFound):
+        arw.add_errors('No result found or multiple results found')
+
+    if arw.has_errors():
+        return arw.to_json(None, 400)
+
+    results = login_schema.dump(login)
+
+    return arw.to_json(results)
+
+
+@login_api_bp.route('/login/<int:login_id>', methods=['PUT'])
+def modify_login(login_id):
+    '''
+    Updates one login object in database
+    '''
+
+    arw = ApiResponseWrapper()
+    login_schema = LoginSchema(exclude=[
+        'updated_at', 'created_at'
+    ])
+    modified_login = request.get_json()
+
+    try:
+        Login.query.filter_by(login_id=login_id).one()
+        modified_login = login_schema.load(modified_login, session=db.session)
+        db.session.commit()
+
+    except (MultipleResultsFound, NoResultFound):
+        arw.add_errors('No result found or multiple results found')
+
+    except ValidationError as ve:
+        arw.add_errors(ve.messages)
+
+    except IntegrityError:
+        arw.add_errors('Integrity error')
+
+    if arw.has_errors():
+        db.session.rollback()
+        return arw.to_json(None, 400)
+
+    results = login_schema.dump(modified_login)
+
+    return arw.to_json(results)
+
+
+@login_api_bp.route('/login', methods=['POST'])
+def check_login_info():
```

Comment:
can we change this to `create_login_info` instead? `check...` seems to imply it's not creating a resource